autoPyTorch/api/base_task.py

```diff
@@ -1,139 +1,740 @@
+                and to validate the performance of a candidate pipeline
+
+        Returns:
+            None
+        """
+        if resampling_strategy is None:
+            raise ValueError("Resampling strategy is needed to determine what models to load")
+        self.ensemble_ = self._backend.load_ensemble(self.seed)
+
+        # If no ensemble is loaded, try to get the best performing model
+        if not self.ensemble_:
+            self.ensemble_ = self._load_best_individual_model()
+
+        if self.ensemble_:
+            identifiers = self.ensemble_.get_selected_model_identifiers()
+            self.models_ = self._backend.load_models_by_identifiers(identifiers)
+            if isinstance(resampling_strategy, CrossValTypes):
+                self.cv_models_ = self._backend.load_cv_models_by_identifiers(identifiers)
+
+            if isinstance(resampling_strategy, CrossValTypes):
+                if len(self.cv_models_) == 0:
+                    raise ValueError('No models fitted!')
+
+        elif 'pipeline' not in self._disable_file_output:
+            model_names = self._backend.list_all_models(self.seed)
+
+            if len(model_names) == 0:
+                raise ValueError('No models fitted!')
+
+            self.models_ = {}
+
+        else:
+            self.models_ = {}
+
+        return True
+
+    def _load_best_individual_model(self) -> SingleBest:
+        """
+        In case of failure during ensemble building,
+        this method returns the single best model found
+        by AutoML.
+        This is a robust mechanism to be able to predict,
+        even though no ensemble was found by ensemble builder.
+        """
+
+        if self._metric is None:
+            raise ValueError("Providing a metric to AutoPytorch is required to fit a model. "
+                             "A default metric could not be inferred. Please check the log "
+                             "for error messages."
+                             )
+
+        # SingleBest contains the best model found by AutoML
+        ensemble = SingleBest(
+            metric=self._metric,
+            seed=self.seed,
+            run_history=self.run_history,
+            backend=self._backend,
+        )
+        if self._logger is None:
+            warnings.warn(
+                "No valid ensemble was created. Please check the log"
+                "file for errors. Default to the best individual estimator:{}".format(
+                    ensemble.identifiers_
+                )
+            )
+        else:
+            self._logger.exception(
+                "No valid ensemble was created. Please check the log"
+                "file for errors. Default to the best individual estimator:{}".format(
+                    ensemble.identifiers_
+                )
+            )
+
+        return ensemble
 
-    @typing.no_type_check
     def fit(
```

Comment:
@ravinkohli  originally had search here, but I recommended him to call it fit, because TPOT/autosklearn/autogluon/Gamma/... uses this interface, so look like almost everyone uses this interface.  But if you got a lot of complains for this nothing more to say :)

Reply:
Yeah, I am actually wondering wether they though about it or just copied it from previous frameworks.. Anyways, `fit` is overloaded, in our case especially since we also have `refit` where fit refers to fitting a specific pipeline and not running the AutoML optimization. I'd say we either call this one `search` or rename `refit` to e.g. `retrain`. I am fine with either option

Reply:
okay, I'll change it to search, so that fit or refit means fitting some pipeline on the given dataset.